core/scripts/common/helpers.go

```diff
@@ -396,9 +403,13 @@ func BinarySearch(top, bottom *big.Int, test func(amount *big.Int) bool) *big.In
 // Get RLP encoded headers of a list of block numbers
 // Makes RPC network call eth_getBlockByNumber to blockchain RPC node
 // to fetch header info
-func GetRlpHeaders(env Environment, blockNumbers []*big.Int) (headers [][]byte, err error) {
+func GetRlpHeaders(env Environment, blockNumbers []*big.Int) (headers [][]byte, hashes []string, err error) {
+
+	hashes = make([]string, len(blockNumbers))
+
 	headers = [][]byte{}
-	for _, blockNum := range blockNumbers {
+	var rlpHeader []byte
```

Comment:
Ah ok, I felt this line was being replicated in all of conditions of the if statement, so pulled it out. As we've more cases in future, this may be replicated more.

`headers = append(headers, rlpHeader)`

Do let me know if you'd like to keep it this way or remove the forward declaration approach